packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/index.ts

```diff
@@ -0,0 +1,52 @@
+// tslint:disable:no-console
+
+import { IsCompleteResponse } from '@aws-cdk/custom-resources/lib/provider-framework/types';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import * as aws from 'aws-sdk';
+import { ClusterResourceHandler, EksClient } from './handler';
+
+aws.config.logger = console;
+
+let eks: aws.EKS | undefined;
+
+const defaultEksClient: EksClient = {
+  createCluster: req => getEksClient().createCluster(req).promise(),
+  deleteCluster: req => getEksClient().deleteCluster(req).promise(),
+  describeCluster: req => getEksClient().describeCluster(req).promise(),
+  updateClusterConfig: req => getEksClient().updateClusterConfig(req).promise(),
+  updateClusterVersion: req => getEksClient().updateClusterVersion(req).promise(),
+  configureAssumeRole: req => {
+    console.log(JSON.stringify({ assumeRole: req }, undefined, 2));
+    const creds = new aws.ChainableTemporaryCredentials({
+      params: req
+    });
+
+    eks = new aws.EKS({ credentials: creds });
+  }
+};
+
+function getEksClient() {
+  if (!eks) {
+    throw new Error(`EKS client not initialized (call "configureAssumeRole")`);
+  }
+
+  return eks;
+}
+
+export async function onEvent(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const provider = new ClusterResourceHandler(defaultEksClient, event);
+  switch (event.RequestType) {
+    case 'Create': return provider.onCreate();
+    case 'Update': return provider.onUpdate();
+    case 'Delete': return provider.onDelete();
+  }
```

Comment:
Hmmm... `ClusterResourceHandler` has the `event`, so it could determine by itself which method should be invoked... Maybe have a single `onEvent()` or `handleEvent()` method instead?

Reply:
Done!